code/controllers/subsystem/radiation.dm

```diff
@@ -28,65 +28,69 @@ SUBSYSTEM_DEF(radiation)
 
 /datum/controller/subsystem/radiation/proc/pulse(atom/source, datum/radiation_pulse_information/pulse_information)
 	var/list/cached_rad_insulations = list()
+	while(pulse_information.turfs_to_process.len)
+		var/turf/turf_to_irradiate = popleft(pulse_information.turfs_to_process)
```

Comment:
if you're gonna iterate turfs_to_process, you should cache it into a local list variable. I'm not joking it's an unnerving amount faster.
Also rather then using popleft, you could count how many things we've iterated, then cut the list right at the end, either when we're done processing or when we yield. S cleaner and a bit cheaper that way.
Also no proc overhead